### ISSUE TITLE
Function setTags in LogClickthrough seems not working

### ISSUE THREAD

**Issue by Messaoudi-Nagib** [elastic/enterprise-search-php#4]
Hi, I want to implement the functionality to get a specific tag when user clicks on a link. The LogClickthrough works without setting a tag, but when I try to set a Tag, it doen't work as expected, here's my code : 

`
        $app = $client->appSearch();
	$szClick = new Request\LogClickthrough($szEngine, $szSearch, $nIdDoc);
	$szClick->setTags(['testTag']);
	// Here's my attempt to put a tag on the logClickthrough response
	$app->logClickthrough($szClick);
`

I've tried different ways to get this work but none of my attempts were successfull

**Comment by ezimuel** [comment]
@Messaoudi-Nagib which versions of `enterprise-search-php` and `Enterprise Search` server are you using?

**Comment by Messaoudi-Nagib** [comment]
Hi, I'm currently using `enterprise-search-php v7.13` and `Enterprise Search v7.13` on the server,

Thanks for the reply

**Comment by ezimuel** [comment]
@Messaoudi-Nagib thanks for the information, the code that you pasted seems correct, I reported here:
```php
$app = $client->appSearch();
$szClick = new Request\LogClickthrough($szEngine, $szSearch, $nIdDoc);
$szClick->setTags(['testTag']); 
// Here's my attempt to put a tag on the logClickthrough response
$response =  $app->logClickthrough($szClick);
```
Some questions:
- which are the values of `$szEngine`, `$szSearch` and `$nIdDoc`?
- what is the error message that are you getting back?
- can you paste the value of `var_dump($response->asString())` and `var_dump($response->getResponse())`?

Thanks!

**Comment by Messaoudi-Nagib** [comment]
Hi, first of all, i've tested if I get any empty values from the variables I've described, and none of these were empty.

Secondly : 

- `$szEngine` is the name of my engine wich is 'import-data'

- `$szSearch` is the query sent by the user to the engine. It's what the user wants to find on the website.

- I've also noticed this type of error on my overview dashboard on App Search : `  "errors": [
    "Contains an invalid tag: {\"0\"=>\"testtag\"}; a tag must be a string"
  ]`  when sometimes my api call passes. It looks like the request is incorrectly formed when I'm adding a tag.

- Var dump of `$response->getResponse` : 

```javascript
object(GuzzleHttp\Psr7\Response)#38 (6) {
  ["reasonPhrase":"GuzzleHttp\Psr7\Response":private]=>
  string(2) "OK"
  ["statusCode":"GuzzleHttp\Psr7\Response":private]=>
  int(200)
  ["headers":"GuzzleHttp\Psr7\Response":private]=>
  array(9) {
    ["X-App-Search-Version"]=>
    array(1) {
      [0]=>
      string(6) "7.13.2"
    }
    ["Content-Type"]=>
    array(1) {
      [0]=>
      string(16) "application/json"
    }
...
```

Response seems to work, I get all the information back.

> 

- ```php var_dump($response->asString()); ``` is giving me blank response ("")

Best regards

**Comment by ezimuel** [comment]
@Messaoudi-Nagib I confirm this is a bug. It seems the `tags` parameter should be sent in the body and not as query string. I'm checking with the API specification. I'll add more details with a fix soon.

**Comment by Messaoudi-Nagib** [comment]
@ezimuel Ok, thank you for the reply :)

**Comment by ezimuel** [comment]
@Messaoudi-Nagib I just sent PR #5 for fixing the `setTags` issue and not only. In fact, this PR fixes also potential other issues for App Search endpoints using query string. With this update, every query string parameter is passed in the body request, as suggested [here](https://www.elastic.co/guide/en/app-search/current/api-reference.html#overview-api-references-parameters).
Let me know if you can try the PR using the new [Request\LogClickthrough](https://github.com/ezimuel/enterprise-search-php/blob/7.13/src/AppSearch/Request/LogClickthrough.php) implementation, thanks!

**Comment by ezimuel** [comment]
@Messaoudi-Nagib I just released enterprise-search-php [7.13.1](https://github.com/elastic/enterprise-search-php/releases/tag/v7.13.1). Let me know if this fix the issue, thanks!

**Comment by Messaoudi-Nagib** [comment]
@ezimuel I've tested locally and it's working now, thank you very much ! Best regards
